src/Microsoft.AspNetCore.Rewrite/ApacheModRewriteOptionsExtensions.cs

```diff
@@ -3,50 +3,45 @@
 
 using System;
 using System.IO;
-using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Rewrite.Internal.ModRewrite;
+using Microsoft.Extensions.FileProviders;
 
 namespace Microsoft.AspNetCore.Rewrite
 {
     /// <summary>
-    /// Apache mod_rewrite extensions on top of the <see cref="RewriteOptions"/>
+    /// Exentsions for adding Apache mod_rewrite rules to <see cref="RewriteOptions"/>
     /// </summary>
     public static class ApacheModRewriteOptionsExtensions
     {
         /// <summary>
-        /// Imports rules from a mod_rewrite file and adds the rules to current rules. 
+        /// Add rules from an Apache mod_rewrite file
         /// </summary>
-        /// <param name="options">The Rewrite options.</param>
-        /// <param name="hostingEnvironment">The Hosting Environment</param>
+        /// <param name="options">The <see cref="RewriteOptions"/></param>
+        /// <param name="fileProvider">The <see cref="IFileProvider"/> </param>
         /// <param name="filePath">The path to the file containing mod_rewrite rules.</param>
-        public static RewriteOptions AddApacheModRewrite(this RewriteOptions options, IHostingEnvironment hostingEnvironment, string filePath)
+        public static RewriteOptions AddApacheModRewrite(this RewriteOptions options, IFileProvider fileProvider, string filePath)
```

Comment:
Open to suggestions. 

> IFileProvider is too generic, it doesn't tell you where to get it.

Regardless, I think we should still provide this API. We can add sugar if we think it's too hard for callers to figure out what to pass in